runners/local-artifact-service-java/src/test/java/org/apache/beam/artifact/local/LocalArtifactStagingLocationTest.java

```diff
@@ -0,0 +1,125 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.artifact.local;
+
+import static com.google.common.base.Preconditions.checkState;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link LocalArtifactStagingLocation}.
+ */
+@RunWith(JUnit4.class)
+public class LocalArtifactStagingLocationTest {
+  @Rule public TemporaryFolder tmp = new TemporaryFolder();
+  @Rule public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void createAtWithAbsentDirectory() throws Exception {
+    File baseFolder = tmp.newFolder();
+    File root = new File(baseFolder, "foo");
+
+    assertThat(root.exists(), is(false));
+    LocalArtifactStagingLocation.createAt(root);
+
+    assertThat(root.exists(), is(true));
+    assertThat(root.listFiles().length, equalTo(1));
+  }
+
+  @Test
+  public void createAtWithExistingDirectory() throws Exception {
+    File root = tmp.newFolder();
+    checkState(root.exists(), "root directory must exist");
+
+    assertThat(root.exists(), is(true));
+    assertThat(root.listFiles().length, equalTo(0));
+    LocalArtifactStagingLocation.createAt(root);
+
+    assertThat(root.exists(), is(true));
+    assertThat(root.listFiles().length, equalTo(1));
+  }
+
+  @Test
+  public void createAtWithUnwritableDirectory() throws Exception {
+    File baseFolder = tmp.newFolder();
+    File root = new File(baseFolder, "foo");
+    checkState(root.mkdir(), "Must be able to create the root directory");
+
+    assertThat(root.exists(), is(true));
+    checkState(root.setWritable(false), "Must be able to set the root directory to unwritable");
+
+    thrown.expect(IllegalStateException.class);
+    LocalArtifactStagingLocation.createAt(root);
+  }
+
+  @Test
+  public void testCreateAtThenForExisting() throws Exception {
+    File baseFolder = tmp.newFolder();
+    LocalArtifactStagingLocation newLocation = LocalArtifactStagingLocation.createAt(baseFolder);
+    File newManifest = newLocation.getManifestFile();
+    assertThat("Manifest creation failed", newManifest.createNewFile(), is(true));
```

Comment:
I would use checkState here instead of assertThat since this is really a precondition to running the test.
You could also use assumeTrue which will mark the test as skipped if the assumption is ever broken but using checkState will force that the test is maintained.

Ditto on other test precondition checks.

assumeTrue works best in cases where you only want to run the test if the system can handle it like only if it can load a JNI library.

Reply:
`checkState` is what we want